### PR TITLE
labhub.py: Block gci issue assignment

### PR DESCRIPTION
**Pull request by EverWinter23**
Block assignment of issues labeled as
'initiatives/gci'.

Closes https://github.com/coala/corobo/issues/501

# Reviewers Checklist

- [ ] Appropriate logging is done.
- [ ] Appropriate error responses.
- [ ] Handle every possible exception.
- [ ] Make sure there is a docstring in the command functions. Hint: Lookout for
  `botcmd` and `re_botcmd` decorators.
- [ ] See that 100% coverage is there.
- [ ] See to it that mocking is not done where it is not necessary.

### CODE REVIEW

File: plugins/labhub.py [hunk 1]
```diff
@@ -301,6 +301,15 @@ def difficulty_level(user, iss):
                         return False
                 elif self.GH3_ORG.is_member(user):
                     return True
+
```
Comment (sils) [inline plugins/labhub.py]: The code does not comply to PEP8.

*Origin: PEP8Bear, Section: `all.autopep8`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpswpgh65g/plugins/labhub.py
+++ b/tmp/tmpswpgh65g/plugins/labhub.py
@@ -301,7 +301,7 @@
                         return False
                 elif self.GH3_ORG.is_member(user):
                     return True
-            
+
             @register_check
             def block_gci_issue_assignment(user, iss):
                 """
```

Comment (sils) [inline plugins/labhub.py]: W293 blank line contains whitespace

*Origin: PycodestyleBear (W293), Section: `all.autopep8`.*

**Comment by EverWinter23** [comment]
Removed a test as the build fails:
https://semaphoreci.com/coala/corobo/branches/pull-request-503/builds/5

Opened another issue for the error and working on the issue:
https://github.com/coala/corobo/issues/502

File: plugins/labhub.py [hunk 2]
```diff
@@ -271,6 +271,16 @@ def register_check(function):
...
+            @register_check
+            def block_gci_issue_assignment(user, iss):
+                """
+                False if the issue is labelled as 'initiatives/gci'.
+                """
+                if 'initiatives/gci' in iss.labels:
+                    return False
+                else:
+                    return True
+
```
Comment (sils) [inline plugins/labhub.py]: Line contains following spacing inconsistencies:
- Trailing whitespaces.

*Origin: SpaceConsistencyBear, Section: `all.whitespace`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmp_tbotk3e/plugins/labhub.py
+++ b/tmp/tmp_tbotk3e/plugins/labhub.py
@@ -280,7 +280,7 @@
                     return False
                 else:
                     return True
-            
+
             @register_check
             def difficulty_level(user, iss):
                 """
```

Comment (sils) [inline plugins/labhub.py]: The code does not comply to PEP8.

*Origin: PEP8Bear, Section: `all.autopep8`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmp_tbotk3e/plugins/labhub.py
+++ b/tmp/tmp_tbotk3e/plugins/labhub.py
@@ -280,7 +280,7 @@
                     return False
                 else:
                     return True
-            
+
             @register_check
             def difficulty_level(user, iss):
                 """
```

Comment (sils) [inline plugins/labhub.py]: The code does not comply to PEP8.

*Origin: PEP8Bear, Section: `all.autopep8`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmp_tbotk3e/plugins/labhub.py
+++ b/tmp/tmp_tbotk3e/plugins/labhub.py
@@ -280,7 +280,7 @@
                     return False
                 else:
                     return True
-            
+
             @register_check
             def difficulty_level(user, iss):
                 """
```

Comment (sils) [inline plugins/labhub.py]: The code does not comply to PEP8.

*Origin: PEP8Bear, Section: `all.autopep8`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmp_tbotk3e/plugins/labhub.py
+++ b/tmp/tmp_tbotk3e/plugins/labhub.py
@@ -280,7 +280,7 @@
                     return False
                 else:
                     return True
-            
+
             @register_check
             def difficulty_level(user, iss):
                 """
```

Comment (sils) [inline plugins/labhub.py]: The code does not comply to PEP8.

*Origin: PEP8Bear, Section: `all.autopep8`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmp_tbotk3e/plugins/labhub.py
+++ b/tmp/tmp_tbotk3e/plugins/labhub.py
@@ -280,7 +280,7 @@
                     return False
                 else:
                     return True
-            
+
             @register_check
             def difficulty_level(user, iss):
                 """
```

Comment (sils) [inline plugins/labhub.py]: The code does not comply to PEP8.

*Origin: PEP8Bear, Section: `all.autopep8`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmp_tbotk3e/plugins/labhub.py
+++ b/tmp/tmp_tbotk3e/plugins/labhub.py
@@ -280,7 +280,7 @@
                     return False
                 else:
                     return True
-            
+
             @register_check
             def difficulty_level(user, iss):
                 """
```

File: plugins/labhub.py [hunk 3]
```diff
@@ -279,12 +279,14 @@ def difficulty_level(user, iss):
                 2. The user belongs to developers or maintainers team as well as
                    newcomers team.
                 False if
-                1. A newcomer asks for assignment to an issue that has no
+                1. A member asks for assignment to an issue that has
```
Comment (sils) [inline plugins/labhub.py]: Line contains following spacing inconsistencies:
- Trailing whitespaces.

*Origin: SpaceConsistencyBear, Section: `all.whitespace`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpapvdlqse/plugins/labhub.py
+++ b/tmp/tmpapvdlqse/plugins/labhub.py
@@ -279,7 +279,7 @@
                 2. The user belongs to developers or maintainers team as well as
                    newcomers team.
                 False if
-                1. A member asks for assignment to an issue that has 
+                1. A member asks for assignment to an issue that has
                    initiatives/gci label.
                 2. A newcomer asks for assignment to an issue that has no
                    difficulty label.
```

File: plugins/labhub.py [hunk 4]
```diff
@@ -279,12 +279,14 @@ def difficulty_level(user, iss):
...
+                   initiatives/gci label.
+                2. A newcomer asks for assignment to an issue that has no
                    difficulty label.
-                2. A newcomer asks for assignment to an issue with difficulty
+                3. A newcomer asks for assignment to an issue with difficulty
                    higher than low.
                 """
-                if (self.TEAMS[self.GH_ORG_NAME + ' newcomers'].is_member(user)
+                elif (self.TEAMS[self.GH_ORG_NAME + ' newcomers'].is_member(user)
                     and not (self.TEAMS[self.GH_ORG_NAME +
```
Comment (sils) [inline plugins/labhub.py]: The code does not comply to PEP8.

*Origin: PEP8Bear, Section: `all.autopep8`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpapvdlqse/plugins/labhub.py
+++ b/tmp/tmpapvdlqse/plugins/labhub.py
@@ -287,8 +287,8 @@
                    higher than low.
                 """
                 elif (self.TEAMS[self.GH_ORG_NAME + ' newcomers'].is_member(user)
-                    and not (self.TEAMS[self.GH_ORG_NAME +
-                                        ' developers'].is_member(
+                      and not (self.TEAMS[self.GH_ORG_NAME +
+                                          ' developers'].is_member(
                             user
                         ) or
                         self.TEAMS[self.GH_ORG_NAME + ' maintainers'].is_member(
```

File: plugins/labhub.py (hunk 3 above)
Comment (sils) [inline plugins/labhub.py]: W291 trailing whitespace

*Origin: PycodestyleBear (W291), Section: `all.autopep8`.*

File: plugins/labhub.py [hunk 5]
```diff
@@ -279,12 +279,14 @@ def difficulty_level(user, iss):
...
+                1. A member asks for assignment to an issue that has 
+                   initiatives/gci label.
+                2. A newcomer asks for assignment to an issue that has no
                    difficulty label.
-                2. A newcomer asks for assignment to an issue with difficulty
+                3. A newcomer asks for assignment to an issue with difficulty
                    higher than low.
                 """
-                if (self.TEAMS[self.GH_ORG_NAME + ' newcomers'].is_member(user)
+                elif (self.TEAMS[self.GH_ORG_NAME + ' newcomers'].is_member(user)
```
Comment (sils) [inline plugins/labhub.py]: E501 line too long (81 > 80 characters)

*Origin: PycodestyleBear (E501), Section: `all.autopep8`.*

File: plugins/labhub.py (hunk 4 above)
Comment (sils) [inline plugins/labhub.py]: E128 continuation line under-indented for visual indent

*Origin: PycodestyleBear (E128), Section: `all.autopep8`.*

File: plugins/labhub.py (hunk 5 above)
Comment (sils) [inline plugins/labhub.py]: Line is longer than allowed. (81 > 80)

*Origin: LineLengthBear, Section: `all.linelength`.*

File: plugins/labhub.py (hunk 4 above)
Comment (sils) [inline plugins/labhub.py]: The code does not comply to PEP8.

*Origin: PEP8Bear, Section: `all.autopep8`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpapvdlqse/plugins/labhub.py
+++ b/tmp/tmpapvdlqse/plugins/labhub.py
@@ -287,8 +287,8 @@
                    higher than low.
                 """
                 elif (self.TEAMS[self.GH_ORG_NAME + ' newcomers'].is_member(user)
-                    and not (self.TEAMS[self.GH_ORG_NAME +
-                                        ' developers'].is_member(
+                      and not (self.TEAMS[self.GH_ORG_NAME +
+                                          ' developers'].is_member(
                             user
                         ) or
                         self.TEAMS[self.GH_ORG_NAME + ' maintainers'].is_member(
```

File: plugins/labhub.py (hunk 3 above)
Comment (sils) [inline plugins/labhub.py]: W291 trailing whitespace

*Origin: PycodestyleBear (W291), Section: `all.autopep8`.*

File: plugins/labhub.py (hunk 5 above)
Comment (sils) [inline plugins/labhub.py]: E501 line too long (81 > 80 characters)

*Origin: PycodestyleBear (E501), Section: `all.autopep8`.*

File: plugins/labhub.py (hunk 4 above)
Comment (sils) [inline plugins/labhub.py]: E128 continuation line under-indented for visual indent

*Origin: PycodestyleBear (E128), Section: `all.autopep8`.*

File: plugins/labhub.py (hunk 5 above)
Comment (sils) [inline plugins/labhub.py]: Line is longer than allowed. (81 > 80)

*Origin: LineLengthBear, Section: `all.linelength`.*

File: plugins/labhub.py (hunk 3 above)
Comment (sils) [inline plugins/labhub.py]: Line contains following spacing inconsistencies:
- Trailing whitespaces.

*Origin: SpaceConsistencyBear, Section: `all.whitespace`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpapvdlqse/plugins/labhub.py
+++ b/tmp/tmpapvdlqse/plugins/labhub.py
@@ -279,7 +279,7 @@
                 2. The user belongs to developers or maintainers team as well as
                    newcomers team.
                 False if
-                1. A member asks for assignment to an issue that has 
+                1. A member asks for assignment to an issue that has
                    initiatives/gci label.
                 2. A newcomer asks for assignment to an issue that has no
                    difficulty label.
```

File: plugins/labhub.py [hunk 6]
```diff
@@ -279,14 +279,18 @@ def difficulty_level(user, iss):
                 2. The user belongs to developers or maintainers team as well as
                    newcomers team.
                 False if
-                1. A newcomer asks for assignment to an issue that has no
+                1. A member asks for assignment to an issue that has
+                   initiatives/gci label.
+                2. A newcomer asks for assignment to an issue that has no
                    difficulty label.
```
Comment (sils) [inline plugins/labhub.py]: The code does not comply to PEP8.

*Origin: PEP8Bear, Section: `all.autopep8`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmp18h0qdt6/plugins/labhub.py
+++ b/tmp/tmp18h0qdt6/plugins/labhub.py
@@ -289,8 +289,8 @@
                 if (self.TEAMS[self.GH_ORG_NAME + ' newcomers'].is_member(
                     user
                     )
-                      and not (self.TEAMS[self.GH_ORG_NAME +
-                                          ' developers'].is_member(
+                    and not (self.TEAMS[self.GH_ORG_NAME +
+                                        ' developers'].is_member(
                             user
                         ) or
                         self.TEAMS[self.GH_ORG_NAME + ' maintainers'].is_member(
```

Comment (sils) [inline plugins/labhub.py]: E127 continuation line over-indented for visual indent

*Origin: PycodestyleBear (E127), Section: `all.autopep8`.*

File: plugins/labhub.py (hunk 4 above)
Comment (sils) [inline plugins/labhub.py]: The code does not comply to PEP8.

*Origin: PEP8Bear, Section: `all.autopep8`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpapvdlqse/plugins/labhub.py
+++ b/tmp/tmpapvdlqse/plugins/labhub.py
@@ -287,8 +287,8 @@
                    higher than low.
                 """
                 elif (self.TEAMS[self.GH_ORG_NAME + ' newcomers'].is_member(user)
-                    and not (self.TEAMS[self.GH_ORG_NAME +
-                                        ' developers'].is_member(
+                      and not (self.TEAMS[self.GH_ORG_NAME +
+                                          ' developers'].is_member(
                             user
                         ) or
                         self.TEAMS[self.GH_ORG_NAME + ' maintainers'].is_member(
```

File: plugins/labhub.py (hunk 3 above)
Comment (sils) [inline plugins/labhub.py]: W291 trailing whitespace

*Origin: PycodestyleBear (W291), Section: `all.autopep8`.*

File: plugins/labhub.py (hunk 5 above)
Comment (sils) [inline plugins/labhub.py]: E501 line too long (81 > 80 characters)

*Origin: PycodestyleBear (E501), Section: `all.autopep8`.*

File: plugins/labhub.py (hunk 4 above)
Comment (sils) [inline plugins/labhub.py]: E128 continuation line under-indented for visual indent

*Origin: PycodestyleBear (E128), Section: `all.autopep8`.*

File: plugins/labhub.py (hunk 5 above)
Comment (sils) [inline plugins/labhub.py]: Line is longer than allowed. (81 > 80)

*Origin: LineLengthBear, Section: `all.linelength`.*

File: plugins/labhub.py (hunk 3 above)
Comment (sils) [inline plugins/labhub.py]: Line contains following spacing inconsistencies:
- Trailing whitespaces.

*Origin: SpaceConsistencyBear, Section: `all.whitespace`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpapvdlqse/plugins/labhub.py
+++ b/tmp/tmpapvdlqse/plugins/labhub.py
@@ -279,7 +279,7 @@
                 2. The user belongs to developers or maintainers team as well as
                    newcomers team.
                 False if
-                1. A member asks for assignment to an issue that has 
+                1. A member asks for assignment to an issue that has
                    initiatives/gci label.
                 2. A newcomer asks for assignment to an issue that has no
                    difficulty label.
```

File: plugins/labhub.py [hunk 7]
```diff
@@ -302,11 +302,27 @@ def difficulty_level(user, iss):
...
+            def block_gci_issue_assignment(user, iss):
+                """
+                True if the issue is not labelled with 'intiatives/gci'.
+                False if the issue has been labelled with 'initiatives.gci'.
+                """
+                if 'initiatives/gci' in iss.labels:
+                    return False
+                else:
+                    return True
+
```
Comment (sils) [inline plugins/labhub.py]: Line contains following spacing inconsistencies:
- Trailing whitespaces.

*Origin: SpaceConsistencyBear, Section: `all.whitespace`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpruo6ink7/plugins/labhub.py
+++ b/tmp/tmpruo6ink7/plugins/labhub.py
@@ -312,7 +312,7 @@
                     return False
                 else:
                     return True
-            
+
         def eligible(user, iss):
             """
             True if all checks are passed.
```

Comment (sils) [inline plugins/labhub.py]: The code does not comply to PEP8.

*Origin: PEP8Bear, Section: `all.autopep8`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpruo6ink7/plugins/labhub.py
+++ b/tmp/tmpruo6ink7/plugins/labhub.py
@@ -312,7 +312,7 @@
                     return False
                 else:
                     return True
-            
+
         def eligible(user, iss):
             """
             True if all checks are passed.
```

Comment (sils) [inline plugins/labhub.py]: W293 blank line contains whitespace

*Origin: PycodestyleBear (W293), Section: `all.autopep8`.*

File: plugins/labhub.py [hunk 8]
```diff
@@ -302,11 +302,27 @@ def difficulty_level(user, iss):
...
+            False if any check(s) is/are failed.
+            """
+            el = True
             for chk in checks:
-                if not chk(user, iss):
-                    return False
-            return True
+                # el evaluates to False if any check is failed.
+                el = el and chk(user, iss):
+            return el
```
Comment (sils) [inline plugins/labhub.py]: E112 expected an indented block

*Origin: PycodestyleBear (E112), Section: `all.autopep8`.*

**Comment by EverWinter23** [comment]
@sils ready for review

File: plugins/labhub.py [hunk 9]
```diff
@@ -278,7 +278,7 @@ def difficulty_level(user, iss):
                 1. A newcomer is asking for assignment to low or newcomer issue.
                 2. The user belongs to developers or maintainers team as well as
                    newcomers team.
-                False if
+                False if:
```
Comment (andrewda) [inline plugins/labhub.py]: Unrelated change; do it in a new commit.

File: plugins/labhub.py [hunk 10]
```diff
@@ -302,11 +302,27 @@ def difficulty_level(user, iss):
...
+            @register_check
+            def block_gci_issue_assignment(user, iss):
+                """
+                True if the issue is not labelled with 'initiatives/gci'.
+                False if the issue has been labelled with 'initiatives.gci'.
+                """
+                if 'initiatives/gci' in iss.labels:
+                    return False
+                else:
+                    return True
```
Comment (andrewda) [inline plugins/labhub.py]: This might also be a good time to make a new way to distinguish GCI members and allow them to work on `initiatives/gci` issues. I think the best way to do this is to create a GCI team in the organization and make sure the user is in that team before assigning them to the issue.

File: plugins/labhub.py [hunk 11]
```diff
@@ -302,11 +302,27 @@ def difficulty_level(user, iss):
...
+            False if any check(s) is/are failed.
+            """
+            el = True
             for chk in checks:
-                if not chk(user, iss):
-                    return False
-            return True
+                # el evaluates to False if any check is failed.
+                el = el and chk(user, iss)
+            return el
```
Comment (andrewda) [inline plugins/labhub.py]: This change to the `eligible` method should be done in a new commit, though I'm not actually sure if I like this or the original better. This is definitely shorter, but also seems to be a little more diluted.

File: plugins/templates/labhub/errors/not-eligible.jinja2.md [hunk 12]
```diff
@@ -1,3 +1,4 @@
 - You must be a member of the {{ organization }} org to be assigned an issue. If you are not a member yet, just type Hello World and corobo will invite you.
 - A newcomer cannot be assigned to an issue with a difficulty level higher than newcomer or low difficulty.
 - A newcomer cannot be assigned to unlabelled issues.
+- initiatives/gci labelled issue assignments are blocked.
```
Comment (andrewda) [inline plugins/templates/labhub/errors/not-eligible.jinja2.md]: I'd say:

```
Only GCI participants can be assigned to issues labelled `initiatives/gci`.
```

**Comment by EverWinter23** [comment]
@andrewda The thing is corobo is unable to assign issues labelled `initiatives/gci` due to https://github.com/coala/corobo/issues/306.

I think this would be more apt:

  corobo cannot assign issues labelled with initiatives/gci.

**Comment by EverWinter23** [comment]
@andrewda @sils ready for review. :grin:

File: plugins/labhub.py [hunk 13]
```diff
@@ -302,11 +302,27 @@ def difficulty_level(user, iss):
...
+
+        def eligible(user, iss):
+            """
+            True if all checks are passed.
+            False if any check(s) is/are failed.
+            """
+            el = True
+            for check in checks:
+                # el evaluates to False if any check is failed.
+                el = el and check(user, iss)
```
Comment (sils) [inline plugins/labhub.py]: why are you changing this method? The other one was more efficient because it didn't run checks if a previous one already failed (that'd be redundand right? `False and <whatever>` will always be false)

**Comment by sils** [comment]
unack https://github.com/coala/corobo/pull/503/commits/6b83dd5a38622d4fafb069fd409e33560ce7caf8

**Comment by EverWinter23** [comment]
@sils Made the requested changes.

File: plugins/labhub.py [hunk 14]
```diff
@@ -302,6 +302,17 @@ def difficulty_level(user, iss):
                 elif self.GH3_ORG.is_member(user):
                     return True
 
+            @register_check
+            def block_gci_issue_assignment(user, iss):
+                """
+                True if the issue is not labelled with 'initiatives/gci'.
+                False if the issue has been labelled with 'initiatives.gci'.
+                """
+                if 'initiatives/gci' in iss.labels:
```
Comment (sils) [inline plugins/labhub.py]: this can be simplified to `return '...' not in iss.labels`

**Comment by meetmangukiya** [comment]
@jayvdb had reverted similar commit from master, I suggest he has a look, and why had he reverted it and do we want this again or not.

**Comment by EverWinter23** [comment]
Will make changes.

**Comment by EverWinter23** [comment]
@sils Pls review. :sweat_smile:

**Comment by EverWinter23** [comment]
@andrewda  Pls review.

**Comment by EverWinter23** [comment]
@Makman2 Review request.

File: plugins/labhub.py (hunk 9 above)
Comment (Makman2) [reply plugins/labhub.py]: bump :+1:

File: plugins/labhub.py [hunk 15]
```diff
@@ -302,6 +302,14 @@ def difficulty_level(user, iss):
                 elif self.GH3_ORG.is_member(user):
                     return True
 
+            @register_check
+            def block_gci_issue_assignment(user, iss):
+                """
+                True if the issue is not labelled with 'initiatives/gci'.
+                False if the issue has been labelled with 'initiatives.gci'.
```
Comment (Makman2) [inline plugins/labhub.py]: Not `initiatives/gci` (with a slash)?

File: tests/labhub_test.py [hunk 16]
```diff
@@ -175,6 +175,25 @@ def test_assign_cmd(self):
 
         testbot.assertCommand(cmd.format('coala', 'a', '23'),
                               'You\'ve been assigned to the issue')
+        
+        # no assignee, newcomer, initiatives/gci
+        mock_maint_team.is_member.return_value = False
+        mock_dev_team.is_member.return_value = False
+        mock_issue.labels = ('initiatives/gci', )
```
Comment (Makman2) [inline tests/labhub_test.py]: Optional: This is equal:
```
mock_issue.labels = 'initiatives/gci',
```

Comment (Makman2) [reply tests/labhub_test.py]: Same below

**Comment by Makman2** [comment]
unack d4b726df8069a885f097a3ad2fdc880eab286a15

**Comment by EverWinter23** [comment]
@Makman2 :+1:

**Comment by EverWinter23** [comment]
Build fails for no reason whatsoever. I keep getting AttributeError, even when the only change that I make is a comment.
Pls look into it.... @meetmangukiya @jayvdb @Makman2 @andrewda @sils 
https://semaphoreci.com/coala/corobo/branches/pull-request-503/builds/25

**Comment by random-access7** [comment]
@EverWinter23 , I think the error _might_ be occurring because your fork is a few commits behind the master branch of coala/corobo and I think it's worth checking out this source of error.

**Comment by EverWinter23** [comment]
@random-access No my fork is not behind the master branch.

**Comment by EverWinter23** [comment]
A dependency repository is causing the build to fail. Waiting for it to be fixed...

**Comment by Makman2** [comment]
--> https://github.com/coala/corobo/pull/503/commits/c2be92c719da887632179a049318230ce88769ee#r175265738

File: tests/labhub_test.py [hunk 17]
```diff
@@ -177,6 +177,27 @@ def test_assign_cmd(self):
...
+        # no assignee, developer, initiatives/gci
+        mock_maint_team.is_member.return_value = False
+        mock_dev_team.is_member.return_value = True
+        mock_issue.labels = ('initiatives/gci', )
+        mock_issue.assignees = tuple()
+        testbot.assertCommand(cmd.format('coala', 'a', '23'),
+                              'not eligible to be assigned to this issue')
+        testbot.pop_message()
+        mock_dev_team.is_member.return_value = False
+        '''
```
Comment (Makman2) [inline tests/labhub_test.py]: why is this commented out? :O

**Comment by Vamshi99** [comment]
>@Vamshi99 ->A dependency repository is causing the build to fail.
Waiting for it to be fixed... 

@EverWinter23 It is fixed, rebase

File: tests/labhub_test.py (hunk 17 above)
Comment (EverWinter23) [reply tests/labhub_test.py]: Sorry.

**Comment by EverWinter23** [comment]
@Makman2 Build still failing. IDK why?

**Comment by Vamshi99** [comment]
@EverWinter23 Your branch is still not same as corobo master. Please rebase

File: plugins/labhub.py (hunk 9 above)
Comment (jayvdb) [reply plugins/labhub.py]: Created a newcomer issue about this https://github.com/coala/corobo/issues/520

File: plugins/templates/labhub/errors/not-eligible.jinja2.md [hunk 18]
```diff
@@ -1,3 +1,4 @@
 - You must be a member of the {{ organization }} org to be assigned an issue. If you are not a member yet, just type Hello World and corobo will invite you.
 - A newcomer cannot be assigned to an issue with a difficulty level higher than newcomer or low difficulty.
 - A newcomer cannot be assigned to unlabelled issues.
+- corobo cannot assign issues labelled with 'initiatives/gci'. Only GCI participants can be assigned these issues.
```
Comment (jayvdb) [inline plugins/templates/labhub/errors/not-eligible.jinja2.md]: This says only GCI participants can be assigned these issues.
However that is not implemented in this patch.

File: plugins/labhub.py [hunk 19]
```diff
@@ -302,6 +302,14 @@ def difficulty_level(user, iss):
                 elif self.GH3_ORG.is_member(user):
                     return True
 
+            @register_check
+            def block_gci_issue_assignment(user, iss):
+                """
+                True if the issue is not labelled with 'initiatives/gci'.
+                False if the issue has been labelled with 'initiatives/gci'.
+                """
+                return 'initiatives/gci' not in iss.labels
```
Comment (jayvdb) [inline plugins/labhub.py]: This doesnt implement allowing GCI participants to assign these issues.

**Comment by EverWinter23** [comment]
@javydb 
Workaround for #306 as it still hasn't been solved and is difficult to implement. This will serve as a short-term solution to the problem.
The solution being:
    gci issues will not be assigned to any contributors using corobo.
    gci members can still claim the issues by commenting on the issue thread.

**Comment by EverWinter23** [comment]
@Makman2 status update?

**Comment by Makman2** [comment]
@EverWinter23 see @jayvdb 's comments ;) Mainly the phrasing inside docs is important :+1:

**Comment by Makman2** [comment]
The workaround blocking those issues for all is okay :+1:

**Comment by EverWinter23** [comment]
@Makman2 Got it. Will Update.

**Comment by EverWinter23** [comment]
@Makman Review request.

File: tests/labhub_test.py (hunk 16 above)
Comment (Makman2) [reply tests/labhub_test.py]: bump

File: plugins/labhub.py (hunk 9 above)
Comment (Makman2) [reply plugins/labhub.py]: bump

**Comment by Makman2** [comment]
Please fix all old comments before requesting a review :+1:

File: plugins/labhub.py (hunk 19 above)
Comment (EverWinter23) [reply plugins/labhub.py]: But, it blocks gci issue assignment using corobo to all users.

**Comment by EverWinter23** [comment]
@Makman2 :sweat_smile: I think I have taken care of everything this time, hopefully.

File: plugins/labhub.py (hunk 19 above)
Comment (bekicot) [reply plugins/labhub.py]: @jayvdb I guess it is good for now, the issue https://github.com/coala/corobo/issues/501 doesn't specify that gci user can assign himself via corobo

File: tests/labhub_test.py [hunk 20]
```diff
@@ -180,6 +180,25 @@ def test_assign_cmd(self):
...
+                              'not eligible to be assigned to this issue')
+        testbot.pop_message()
+
+        # no assignee, developer, initiatives/gci
+        mock_maint_team.is_member.return_value = False
+        mock_dev_team.is_member.return_value = True
+        mock_issue.labels = 'initiatives/gci',
+        mock_issue.assignees = tuple()
+        testbot.assertCommand(cmd.format('coala', 'a', '23'),
+                              'not eligible to be assigned to this issue')
```
Comment (Makman2) [inline tests/labhub_test.py]: Where is the `You are ...` in the message? :) I would compare against the full message, so nobody asks you why you are asserting for just a part of the message (like me) ;)

Comment (EverWinter23) [reply tests/labhub_test.py]: @Makman2 Added the 'You are ...' :smiley:

File: tests/labhub_test.py [hunk 21]
```diff
@@ -180,6 +180,27 @@ def test_assign_cmd(self):
...
+        testbot.pop_message()
+
+        # no assignee, developer, initiatives/gci
+        mock_maint_team.is_member.return_value = False
+        mock_dev_team.is_member.return_value = True
+        mock_issue.labels = 'initiatives/gci',
+        mock_issue.assignees = tuple()
+        testbot.assertCommand(cmd.format('coala', 'a', '23'),
+                              'You are not eligible to be assigned\
+                               to this issue')
```
Comment (nvzard) [inline tests/labhub_test.py]: @EverWinter23 use
```
 testbot.assertCommand(cmd.format('coala', 'a', '23'),
                                  'You are not eligible to be assigned ' \
                                  'to this issue')
```

Comment (EverWinter23) [reply tests/labhub_test.py]: @nvzard @Makman2 Done

Comment (nvzard) [reply tests/labhub_test.py]: @EverWinter23 not really. You did a silly mistake. Make the builds happy ;)

**Comment by EverWinter23** [comment]
@Makman2 @nvzard All done.

File: tests/labhub_test.py [hunk 22]
```diff
@@ -180,6 +180,27 @@ def test_assign_cmd(self):
...
         testbot.assertCommand(cmd.format('coala', 'a', '23'),
                               'You\'ve been assigned to the issue')
+        
+        # no assignee, newcomer, initiatives/gci
+        mock_maint_team.is_member.return_value = False
+        mock_dev_team.is_member.return_value = False
+        mock_issue.labels = 'initiatives/gci',
+        mock_issue.assignees = tuple()
+        testbot.assertCommand(cmd.format('coala', 'a', '23'),
+                              'You are not eligible to be assigned ' \
```
Comment (jayvdb) [inline tests/labhub_test.py]: unnecessary `\`
python has implicit string concatenation

Comment (EverWinter23) [reply tests/labhub_test.py]: Thanks, I'm don't know how you know things like this. I get to learn so much from you guys. This is the best community out there. You guys are the best. @jayvdb @Makman2 @nvzard

**Comment by EverWinter23** [comment]
@jayvdb Done.

**Comment by jayvdb** [comment]
ack https://github.com/coala/corobo/pull/503/commits/0f8a0c0f89c7f14edb14bf2c15a132b5476bb841

**Comment by jayvdb** [comment]
@gitmate-bot ff
